src/OpenSage.Game/Logic/AI/AIStates/StopHackingInternetState.cs

```diff
@@ -4,13 +4,54 @@ namespace OpenSage.Logic.AI.AIStates
 {
     internal sealed class StopHackingInternetState : State
     {
-        public LogicFrameSpan FramesUntilFinishedPacking;
+        public const uint Id = 1002;
+
+        private readonly GameObject _gameObject;
+
+        private LogicFrameSpan _framesUntilFinishedPacking;
+
+        public StopHackingInternetState(GameObject gameObject)
+        {
+            _gameObject = gameObject;
+        }
+
+        public override void OnEnter()
+        {
+            _gameObject.ModelConditionFlags.Set(ModelConditionFlag.Unpacking, false);
+            _gameObject.ModelConditionFlags.Set(ModelConditionFlag.FiringA, false);
+            _gameObject.ModelConditionFlags.Set(ModelConditionFlag.Packing, true);
+
+            _gameObject.GameContext.AudioSystem.PlayAudioEvent(_gameObject, _gameObject.Definition.UnitSpecificSounds.UnitPack?.Value);
+
+            var aiUpdate = (HackInternetAIUpdate)_gameObject.AIUpdate;
+
+            var frames = aiUpdate.GetVariableFrames(aiUpdate.ModuleData.PackTime);
```

Comment:
It would be nice if we could pass this in (or perhaps just expose a Set method or similar? really it would be nice if we could _construct_ this state with this parameter) so we don't have to access the AIUpdate object from here, and then `ModuleData` and `GetVariableFrames` could remain private. However, given the way we currently set states, that could be a more complex task.

The same goes for StartHackingInternetState as well. Perhaps some sort of `transitionTimer` or similar could be added and utilized for other modules as well?

Reply:
> Perhaps some sort of transitionTimer or similar could be added and utilized for other modules as well?

That could be nice, but I think I'd prefer to implement a few more modules and then see what common behaviour we want to extract into helper classes. I just don't know yet what we'll need.